spec/webhooks/packerpipeline_spec.rb

```diff
@@ -0,0 +1,84 @@
+require 'rspec'
+require 'rspec/mocks'
+require 'tempfile'
+require 'pathname'
+require_relative '../../files/default/lib/packerpipeline'
+
+module SpecHelper
+  def fixture_path(name)
+    File.join(File.expand_path('../../fixtures', __FILE__), name)
+  end
+
+  def open_fixture(name)
+    File.read(fixture_path(name))
+  end
+
+  def open_json(name)
+    JSON.parse(open_fixture(name))
+  end
+
+  def tempfile(file)
+    tempfile = Tempfile.new("#{Pathname.new(file).basename}-rspec")
+    tempfile.write(::File.read(file))
+    tempfile.rewind
+    tempfile
+  end
+end
+
+RSpec.configure do |c|
+  c.include SpecHelper
+end
+
+describe PackerPipeline do
+  context '#changed_files' do
+    let(:github_mock) { double('Octokit', commits: [], issues: [], same_options?: false, auto_paginate: true) }
+    before :each do
+      allow(Octokit::Client).to receive(:new) { github_mock }
+    end
+    it 'finds single changed file' do
```

Comment:
The test should rather be restructured as

1. "finds changed template"
1. "finds a changed script"
1. "finds changed template and a changed script".

The `PR#`, and `arch` output test are good and should stay.
The arch output test could be expanded to process and output only one of the `arch`s at a time and then both.